Validation/MtdValidation/plugins/MtdTracksValidation.cc

```diff
@@ -49,47 +51,78 @@ class MtdGlobalRecoValidation : public DQMEDAnalyzer {
   const float trackMinEta_;
   const float trackMaxEta_;
 
+  edm::EDGetTokenT<reco::TrackCollection> GenRecTrackToken_;
   edm::EDGetTokenT<reco::TrackCollection> RecTrackToken_;
   edm::EDGetTokenT<std::vector<reco::Vertex>> RecVertexToken_;
 
+  edm::EDGetTokenT<edm::ValueMap<int>> trackAssocToken_;
+  edm::EDGetTokenT<edm::ValueMap<float>> pathLengthToken_;
+
+  edm::EDGetTokenT<edm::ValueMap<float>> tmtdToken_;
+  edm::EDGetTokenT<edm::ValueMap<float>> SigmatmtdToken_;
+  edm::EDGetTokenT<edm::ValueMap<float>> t0PidToken_;
+  edm::EDGetTokenT<edm::ValueMap<float>> Sigmat0PidToken_;
+  edm::EDGetTokenT<edm::ValueMap<float>> t0SafePidToken_;
+  edm::EDGetTokenT<edm::ValueMap<float>> Sigmat0SafePidToken_;
+  edm::EDGetTokenT<edm::ValueMap<float>> trackMVAQualToken_;
+
   MonitorElement* meBTLTrackRPTime_;
   MonitorElement* meBTLTrackEffEtaTot_;
   MonitorElement* meBTLTrackEffPhiTot_;
   MonitorElement* meBTLTrackEffPtTot_;
   MonitorElement* meBTLTrackEffEtaMtd_;
   MonitorElement* meBTLTrackEffPhiMtd_;
   MonitorElement* meBTLTrackEffPtMtd_;
+  MonitorElement* meBTLTrackPtRes_;
 
-  MonitorElement* meETLTrackRPTime_[4];
-  MonitorElement* meETLTrackNumHits_[4];
+  MonitorElement* meETLTrackRPTime_;
   MonitorElement* meETLTrackEffEtaTot_[2];
   MonitorElement* meETLTrackEffPhiTot_[2];
   MonitorElement* meETLTrackEffPtTot_[2];
   MonitorElement* meETLTrackEffEtaMtd_[2];
   MonitorElement* meETLTrackEffPhiMtd_[2];
   MonitorElement* meETLTrackEffPtMtd_[2];
-
+  MonitorElement* meETLTrackPtRes_;
+
+  MonitorElement* meTracktmtd_;
+  MonitorElement* meTrackSigmatmtd_;
+  MonitorElement* meTrackt0Pid_;
+  MonitorElement* meTrackSigmat0Pid_;
+  MonitorElement* meTrackt0SafePid_;
+  MonitorElement* meTrackSigmat0SafePid_;
   MonitorElement* meTrackNumHits_;
+  MonitorElement* meTrackMVAQual_;
+  MonitorElement* meTrackPathLenghtvsEta_;
 
   MonitorElement* meVerNumber_;
   MonitorElement* meVerZ_;
   MonitorElement* meVerTime_;
 };
 
 // ------------ constructor and destructor --------------
-MtdGlobalRecoValidation::MtdGlobalRecoValidation(const edm::ParameterSet& iConfig)
+MtdTracksValidation::MtdTracksValidation(const edm::ParameterSet& iConfig)
     : folder_(iConfig.getParameter<std::string>("folder")),
-      trackMinEnergy_(iConfig.getParameter<double>("trackMinimumEnergy")),
+      trackMinEnergy_(iConfig.getParameter<double>("trackMinimumPt")),
```

Comment:
it would be more clear to have the local variable coherently mentioning pt instead of energy, although the code does not change its behaviour